skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/aibot/ChatWindow.kt

```diff
@@ -0,0 +1,133 @@
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import org.jetbrains.jewel.foundation.theme.JewelTheme
+import org.jetbrains.jewel.ui.component.Icon
+import org.jetbrains.jewel.ui.component.IconButton
+import org.jetbrains.jewel.ui.component.Text
+import org.jetbrains.jewel.ui.component.TextArea
+
+@Composable
+fun ChatWindow(modifier: Modifier = Modifier) {
+  Column(
+    modifier = Modifier.fillMaxSize().background(JewelTheme.globalColors.paneBackground),
+    verticalArrangement = Arrangement.Bottom,
+  ) {
+    ConversationField(modifier)
+  }
+}
+
+@Composable
+fun ConversationField(modifier: Modifier = Modifier) {
+  var textValue by remember { mutableStateOf(TextFieldValue()) }
+  val isTextNotEmpty = textValue.text.isNotBlank()
+  Row(
+    modifier.fillMaxWidth().height(IntrinsicSize.Min).padding(4.dp),
+    horizontalArrangement = Arrangement.Center,
+    verticalAlignment = Alignment.CenterVertically,
+  ) {
+    TextArea(
+      value = textValue,
+      onValueChange = { newText -> textValue = newText },
+      modifier =
+        Modifier.weight(1f).padding(4.dp).heightIn(min = 56.dp).onPreviewKeyEvent { event ->
+          when {
+            (event.key == Key.Enter || event.key == Key.NumPadEnter) &&
+              event.type == KeyEventType.KeyDown -> {
+              if (event.isShiftPressed) {
+                val newText =
+                  textValue.text.replaceRange(
+                    textValue.selection.start,
+                    textValue.selection.end,
+                    "\n",
+                  )
+                val newSelection = TextRange(textValue.selection.start + 1)
+                textValue = TextFieldValue(newText, newSelection)
+                true
+              } else {
+                textValue = TextFieldValue("")
+                true
+              }
+            }
+            else -> false
+          }
+        },
+      placeholder = { Text("Start your conversation") },
+      keyboardActions = KeyboardActions.Default,
+      maxLines = Int.MAX_VALUE,
+    )
+    Column(Modifier.fillMaxHeight(), verticalArrangement = Arrangement.Center) {
+      Row(verticalAlignment = Alignment.CenterVertically) {
+        IconButton(
+          modifier = Modifier.padding(4.dp).fadeWhenDisabled(isTextNotEmpty),
+          onClick = {
+            if (isTextNotEmpty) {
+              textValue = TextFieldValue()
+            }
+          },
+          enabled = isTextNotEmpty,
+        ) {
+          Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.spacedBy(4.dp),
+          ) {
+            Icon(
+              painter = painterResource("/drawable/send.svg"),
+              contentDescription = "Send",
+              modifier = Modifier.size(20.dp),
+            )
+          }
+        }
+      }
+    }
+  }
+}
+
+fun Modifier.fadeWhenDisabled(enabled: Boolean): Modifier {
```

Comment:
Fade is usually an animation, but this isn't in this case. Let's call it something like `enabled` and make it more about its semantic meaning.

Also, material has a pattern for this we could follow from its `ContentAlpha` class - https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/material/material/src/commonMain/kotlin/androidx/compose/material/ContentAlpha.kt;l=58?q=ContentAlpha

Reply:
Ah okay, yeah this makes sense! I will add this adjustment to the next PR then.